vm/src/builtins/weakproxy.rs

```diff
@@ -70,6 +71,16 @@ impl PyWeakProxy {
         })?;
         obj.get_attr(attr_name, vm)
     }
+    #[pymethod(magic)]
+    fn str(&self, vm: &VirtualMachine) -> PyResult<PyStrRef> {
+        match self.weak.upgrade() {
+            Some(obj) => obj.str(vm),
+            None => Err(vm.new_exception_msg(
+                vm.ctx.exceptions.reference_error.clone(),
+                "weakly-referenced object no longer exists".to_owned(),
+            )),
```

Comment:
this part is an exact repeat of error in `getattr()`.

please create a function `new_reference_error()` not to manually repeat the exception creation.